scripts/setup

```diff
@@ -0,0 +1,37 @@
+# source this script to prepare some common environment variables
+
+# adapted from https://github.com/johannes-wolf/cetz/blob/35c0868378cea5ad323cc0d9c2f76de8ed9ba5bd/scripts/package
+# licensed under Apache License 2.0
+
+# Local package directories per platform
+if [[ "$OSTYPE" == "linux"* ]]; then
+  DATA_DIR="${XDG_DATA_HOME:-$HOME/.local/share}"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  DATA_DIR="$HOME/Library/Application Support"
+else
+  DATA_DIR="${APPDATA}"
+fi
+
+function read-toml() {
+  local file="$1"
+  local key="$2"
+  # Read a key value pair in the format: <key> = "<value>"
+  # stripping surrounding quotes.
+  perl -lne "print \"\$1\" if /^${key}\\s*=\\s*\"(.*)\"/" < "$file"
+}
```

Comment:
I can't say I'm a fan of having hardly readable bash scripts in such a template. I know what it does, but I can already tell that it's flaky, considering that it cannot handle flat toml files which may be generated automatically somewhere.

This is also hardly portable, as perl is not really a thing you find on the average Windows PC.

Reply:
yeah, this came directly from [CeTZ](https://github.com/cetz-package/common/blob/17b94406166baa0d14a65326bb260aebca34176f/scripts/package) without touching it...